File: leetcode.com/0203_remove_linked_list_elements/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        node = head

        # skip begin of target
        while node and node.val == val:
            node = node.next

        nh = node
        prev = node

        while node:

            if node.val == val:
                prev.next = node.next
            else:
                prev = node

            node = node.next

        return nh

File: leetcode.com/0203_remove_linked_list_elements/solution.c
struct ListNode* removeElements(struct ListNode* head, int val)
{
    while (head && head->val == val)
        head = head->next;

    struct ListNode *node = head;

    while (node) {
        if (node->next && node->next->val == val)
            node->next = node->next->next;
        else
            node = node->next;
    }

    return head;
}
